core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/deploysemantics/PhysicalOp.scala

```diff
@@ -468,8 +474,10 @@ case class PhysicalOp(
     * Tells whether the input on this link is blocking i.e. the operator doesn't output anything till this link
     * outputs all its tuples
     */
-  def isInputLinkBlocking(link: PhysicalLink): Boolean = {
-    val blockingLinks = realBlockingInputs.flatMap(portId => getInputLinks(Some(portId)))
+  def isLinkBlocking(link: PhysicalLink): Boolean = {
```

Comment:
I suggest define `blocking` as a property of the output port. and remove all functions assessing blocking on links. note that you can always use `link.fromPortId` to find the port, and then find its property.

Reply:
Per our discussion, let's keep this for now.